x-pack/plugins/apm/public/components/alerting/ui_components/apm_rule_params_container/index.tsx

```diff
@@ -41,8 +68,63 @@ export function ApmRuleParamsContainer(props: Props) {
           </EuiFlexItem>
         ))}
       </EuiFlexGrid>
+
       {chartPreview}
       <EuiSpacer size="m" />
     </>
   );
 }
+
+function MinimumWindowSizeWarning({
+  minimumWindowSize,
+}: {
+  minimumWindowSize: MinimumWindowSize;
+}) {
+  return (
+    <EuiCallOut
+      title={`Please increase "For the last" to at least ${
+        minimumWindowSize.value
+      } ${getTimeUnitLabel(minimumWindowSize.unit)}`}
+      color="warning"
+      iconType="alert"
+    >
+      <p>
+        The recommended minimum value is {minimumWindowSize.value}{' '}
+        {getTimeUnitLabel(minimumWindowSize.unit)}. This is to ensure that the
+        alert has enough data to evaluate. If you choose a value that is too
+        low, the alert may not fire as expected.
+      </p>
+    </EuiCallOut>
+  );
+}
+
+function useShowMinimumWindowSize({
```

Comment:
Does it have to be a hook at all? It looks like a normal function that returns true/false?!

Reply:
Are you suggesting to inlining this? I think it's better to extract it but I don't feel strongly about it.